### DISCUSSION TITLE
can't register which-key group

### DISCUSSION THREAD

**Discussion by CaptainJack42** [LazyVim/LazyVim#179] [Q&A] [answered]
Hey,

i recently started using this awesome Project and am currently trying to port some of my mappings and register them with which-key, but i can't seem to be able to register a new group with which-key. 

For example i have some bindings registered in `lua/config/keymaps.lua` for the Rust plugin i'm using like this: 

```
local function map(mode, lhs, rhs, opts)
  local keys = require("lazy.core.handler").handlers.keys
  ---@cast keys LazyKeysHandler
  -- do not create the keymap if a lazy keys handler exists
  if not keys.active[keys.parse({ lhs, mode = mode }).id] then
    vim.keymap.set(mode, lhs, rhs, opts)
  end
end

-- Rust keybinds
map("n", "<leader>rr", ":RustRun <CR>", { desc = "Rust run" })
map("n", "<leader>ra", ":RustRunnables <CR>", { desc = "Rust runnables" })
map("n", "<leader>rd", ":RustDebuggables <CR>", { desc = "Rust debuggables" })
map("n", "<leader>rk", ":RustMoveItemUp <CR>", { desc = "Rust move item up" })
map("n", "<leader>rj", ":RustMoveItemDown <CR>", { desc = "Rust move item down" })
map("n", "<leader>rh", ":RustHoverActions <CR>", { desc = "Rust Hover Actions (help)" })
```

I'm now trying to register the group `+rust` with which-key, following [this](https://github.com/folke/which-key.nvim#%EF%B8%8F-mappings) setup. I created the file `lua/config/which-key.lua` containing this:

```
local wk = require("which-key")

wk.register({
  ["<leader>"] = {
    r = {
      name = "+rust",
    },
  },
})
```

> The files under lua/config will be automatically loaded at the appropriate time, so you don't need to require those files manually.

From this section in the [documentation](https://www.lazyvim.org/configuration/general) i'd suspect that the config file i created should be automatically loaded, however this doesn't seem to be the case, when pressing `<leader>` the name for `r` is still `+prefix`.

**Comment by folke** [comment, marked as answer]
Only the files mentioned would be sourced, not random files

**Reply by dpetka2001** [reply]
If you move your file to `plugins` directory I think it will be sourced. I also have a `which-key.lua` file in my `plugins` directory 
to add some extra groups and it registers them correctly.
